### ISSUE TITLE
Single file mode with `path` and `version` dependencies, doesn't compile versioned libraries

### ISSUE THREAD

**Issue by wilzbach** [dlang/dub#893]
Hey I bumped into an issue with the single file mode with https://github.com/libmir/mir/pull/250

Problem: If one dependency is given as `path` and they other with `version`, the library won't be compiled correctly.

The file looks roughly like this:

```d
#!/usr/bin/env dub
/+ dub.sdl:
name "median_filter"
dependency "mir" path=".."
dependency "imageformats" version="~>6.0.0"
+/
```

And I get the following error. Interestingly when rerunning the command, it properly works.

```
ubuntu@box328:~/mir$ dub build -v --single examples/median_filter.d 
Using dub registry url 'http://code.dlang.org/'
Refreshing local packages (refresh existing: true)...
Looking for local package map at /var/lib/dub/packages/local-packages.json
Looking for local package map at /home/ubuntu/.dub/packages/local-packages.json
Determined package version using GIT: mir ~use-local-paths
Refreshing local packages (refresh existing: false)...
Looking for local package map at /var/lib/dub/packages/local-packages.json
Looking for local package map at /home/ubuntu/.dub/packages/local-packages.json
  Version selection for dependency mir (mir) of median_filter is missing.
  Adding local /home/ubuntu/mir
  Found dependency mir >=0.0.0 @..
  Version selection for dependency imageformats (imageformats) of median_filter is missing.
  Missing dependency imageformats ~>6.0.0 of median_filter
Refreshing local packages (refresh existing: false)...
Looking for local package map at /var/lib/dub/packages/local-packages.json
Looking for local package map at /home/ubuntu/.dub/packages/local-packages.json
  Version selection for dependency mir (mir) of median_filter is missing.
  Adding local /home/ubuntu/mir
  Found dependency mir >=0.0.0 @..
  Version selection for dependency imageformats (imageformats) of median_filter is missing.
  Missing dependency imageformats ~>6.0.0 of median_filter
Checking for missing dependencies.
Search for versions of mir (1 package suppliers)
Return for mir: [0.15.3, 0.15.2, 0.13.3, 0.13.2, 0.13.1, 0.13.0, 0.11.1, 0.10.2, 0.15.1-beta2, 0.15.1-beta, 0.15.0-beta4, 0.15.0-beta3, 0.15.0-beta2, 0.15.0-beta, 0.12.0-beta, 0.11.0-beta2, 0.11.0-beta, 0.10.1-alpha, 0.10.0-beta, 0.10.0-alpha, 0.9.0-beta.1, 0.9.0-beta, 0.9.0-alpha.1, 0.9.0-alpha.0, ~master, ~wilzbach-patch-2, ~use-local-paths, ~tinflex, ~blas, ~9il-patch-3]
Search for versions of imageformats (1 package suppliers)
Return for imageformats: [6.0.0, 5.2.0, 5.1.0, 5.0.0, 4.2.2, 4.2.1, 4.2.0, 4.1.1, 4.1.0, 4.0.1, 4.0.0, 3.1.0, 3.0.6, 3.0.5, 3.0.4, 3.0.3, 2.0.1, 2.0.0, 1.0.0, ~master]
Fetching imageformats 6.0.0 (getting selected version)...
Acquiring package zip file
Found download URL: 'http://code.dlang.org/packages/imageformats/6.0.0.zip'
Placing imageformats 6.0.0 to /home/ubuntu/.dub/packages/...
Placing package 'imageformats' version '6.0.0' to location '/home/ubuntu/.dub/packages/imageformats-6.0.0/imageformats' from file '/tmp/imageformats-314039ed-a2ac-4ad1-86cc-ab8a03a367c7.zip'
Copying all files...
41 file(s) copied.
Refreshing local packages (refresh existing: false)...
Looking for local package map at /var/lib/dub/packages/local-packages.json
Looking for local package map at /home/ubuntu/.dub/packages/local-packages.json
  Found dependency mir >=0.0.0 @..
  Found dependency imageformats 6.0.0
Generating using build
Generate target median_filter (executable /home/ubuntu/mir/examples median_filter)
Generate target imageformats (staticLibrary /home/ubuntu/.dub/packages/imageformats-6.0.0/imageformats imageformats)
Generate target mir (sourceLibrary /home/ubuntu/mir mir)
Performing "debug" build using dmd for x86_64.
File '../.dub/packages/imageformats-6.0.0/imageformats/imageformats/bmp.d' modified, need rebuild.
imageformats 6.0.0: building configuration "library"...
dmd -lib -of../../../tmp/.dub/build/imageformats-6.0.0/library-debug-linux.posix-x86_64-dmd_2071-B46E74BFBCD14A19AF176FFDBADE5C6B/libimageformats.a -debug -g -w -version=Have_imageformats -I../.dub/packages/imageformats-6.0.0/imageformats ../.dub/packages/imageformats-6.0.0/imageformats/imageformats/bmp.d ../.dub/packages/imageformats-6.0.0/imageformats/imageformats/jpeg.d ../.dub/packages/imageformats-6.0.0/imageformats/imageformats/package.d ../.dub/packages/imageformats-6.0.0/imageformats/imageformats/png.d ../.dub/packages/imageformats-6.0.0/imageformats/imageformats/tga.d -vcolumns
File ../.dub/packages/imageformats-6.0.0/imageformats/libimageformats.a doesn't exists, triggering rebuild.
median_filter ~master: building configuration "application"...
dmd -c -of../../../tmp/.dub/build/median_filter-~master/application-debug-linux.posix-x86_64-dmd_2071-5BDC9A744C4986AE20A77CB791A49B03/median_filter.o -debug -g -w -version=Have_median_filter -version=Have_mir -version=Have_imageformats -I../.dub/packages/imageformats-6.0.0/imageformats -Isource/ examples/median_filter.d source/mir/blas/dot.d source/mir/blas/gemm.d source/mir/blas/gemv.d source/mir/blas/package.d source/mir/combinatorics/package.d source/mir/internal/math.d source/mir/internal/utility.d source/mir/math.d source/mir/model/lda/hoffman.d source/mir/ndslice/internal.d source/mir/ndslice/iteration.d source/mir/ndslice/package.d source/mir/ndslice/selection.d source/mir/ndslice/slice.d source/mir/random/discrete.d source/mir/sparse/blas/axpy.d source/mir/sparse/blas/dot.d source/mir/sparse/blas/gemm.d source/mir/sparse/blas/gemv.d source/mir/sparse/blas/package.d source/mir/sparse/package.d source/mir/sum.d -vcolumns
Linking...
dmd -of../../../tmp/.dub/build/median_filter-~master/application-debug-linux.posix-x86_64-dmd_2071-5BDC9A744C4986AE20A77CB791A49B03/median_filter ../../../tmp/.dub/build/median_filter-~master/application-debug-linux.posix-x86_64-dmd_2071-5BDC9A744C4986AE20A77CB791A49B03/median_filter.o ../.dub/packages/imageformats-6.0.0/imageformats/libimageformats.a -L--no-as-needed -g
cc: error: ../.dub/packages/imageformats-6.0.0/imageformats/libimageformats.a: No such file or directory
--- errorlevel 1
FAIL ../../../tmp/.dub/build/median_filter-~master/application-debug-linux.posix-x86_64-dmd_2071-5BDC9A744C4986AE20A77CB791A49B03/ median_filter executable
dmd failed with exit code 1.
```

Dub version 1.0.0

**Comment by wilzbach** [comment]
Update turns out that it has nothing to do with `path`, it's probably more cache related.

```
rm -rf ~/.dub
./test.d
```

`test.d`

```d
!/usr/bin/env dub
/+ dub.sdl:
name "test"
dependency "mir" version=">0.15.1"
dependency "imageformats" version="~>6.0.0"
+/

import mir.ndslice;
void main(string[] args)
{
    import std.stdio;
    writeln("test");
}
```

will print the same error.

**Comment by s-ludwig** [comment]
Fixed by #922 (#921 was a dup.)
